packages/financial-templates-lib/src/helpers/GasEstimator.ts

```diff
@@ -21,43 +21,54 @@ interface GasEstimatorMapping {
     backupUrl?: string;
   };
 }
-
-interface EtherchainResponse {
-  safeLow: number;
-  standard: number;
-  fast: number;
-  fastest: number;
-  currentBaseFee: number;
-  recommendedBaseFee: number;
+interface EtherscanGasResponse {
+  status: string;
+  message: string;
+  result: {
+    LastBlock: string;
+    SafeGasPrice: string;
+    ProposeGasPrice: string;
+    FastGasPrice: string;
+    suggestBaseFee: string;
+    gasUsedRatio: string;
+  };
 }
 
+interface MaticGasPriceData {
+  maxPriorityFee: number | string;
+  maxFee: number | string;
+}
 interface MaticResponse {
-  safeLow: number;
-  standard: number;
-  fast: number;
-  fastest: number;
+  safeLow: MaticGasPriceData;
+  standard: MaticGasPriceData;
+  fast: MaticGasPriceData;
+  estimatedBaseFee: string;
   blockTime: number;
   blockNumber: number;
 }
 
 export const MAPPING_BY_NETWORK: GasEstimatorMapping = {
-  // Expected shape:
-  // <netId>: {
-  //     url: <primary-gas-station-url>,
-  //     backupUrl: <optional-backup-gas-station-url>,
-  //     defaultFastPricesGwei: <default-gas-price-for-network>
-  // }
   1: {
-    url: "https://www.etherchain.org/api/gasPriceOracle",
-    defaultMaxFeePerGasGwei: 500,
-    defaultMaxPriorityFeePerGasGwei: 5,
+    url: "https://api.etherscan.io/api?module=gastracker&action=gasoracle",
+    defaultMaxFeePerGasGwei: 450, // maxFeePerGas = baseFeePerGas + maxPriorityFeePerGas
+    defaultMaxPriorityFeePerGasGwei: 100,
     type: NetworkType.London,
   },
   10: { defaultFastPriceGwei: 1, type: NetworkType.Legacy },
-  137: { url: "https://gasstation-mainnet.matic.network", defaultFastPriceGwei: 10, type: NetworkType.Legacy },
+  137: {
+    url: "https://gasstation.polygon.technology/v2",
```

Comment:
If we are using api.etherscan.io for mainnet, why just not use api.polygonscan.com here if this would allow unifying response interfaces?

Reply:
Makes sense!

However https://api-testnet.polygonscan.com/ doenst have `gastracker` 

https://docs.polygonscan.com/v/mumbai-polygonscan/

Reply:
Added in last commit